### DISCUSSION TITLE
A few suggestions

### DISCUSSION THREAD

**Discussion by FlorianH27** [Mattk70/Chirpity-Electron#42] [Ideas]
Hello
Firstly, I am impressed with Chirpity NocMig and its unique features for NocMig! 
I have some suggestions that may be implemented in the future:
- The ability to scroll through the spectrogram with a mouse or a trackpad to improve the navigation. 
- It would be helpful to be able to adjust the playback volume in Chirpity, as my NocMig files are often very quiet. My computer's maximum volume is usually insufficient, and I would have to manually increase the volume of each file separately. (Until now I have always analysed my audio files in Audacity, where the volume can be adjusted. But I don't know if others have the problem too.)
- It would be great if it was possible to train one's own Chirpity or BirdNET model. For example, if the model could learn from my feedback when I set the species with "Edit Record". Perhaps an easier solution would be to allow the import of self-trained BirdNET models.

Best regards,
Florian Hatt

**Comment by Vollpflock** [comment]
Ciao Florian

Nice to hear from you! I can't help, but I have made two similar feature requests. 

For 1, see https://github.com/Mattk70/Chirpity-Electron/issues/21 Nr. 10.  Scrolling would be nice though, I agree
For 2, see https://github.com/Mattk70/Chirpity-Electron/issues/28.

grüessli

**Comment by Mattk70** [comment]
Hi, thanks for getting in touch, Florian. 

As @Vollpflock says, and the link will highlight, there are keyboard shortcuts, like PageUp / Page down to quickly navigate the Spectrogram. The mouse /trackpad suggestion is a great idea, but I can make no promises! There are so many things higher up the list!

One of those is a Gain node, suggested in #28 . I don't imagine this will be too hard, but I need to think about how to fit it into the UI, I don't think it belongs with the other "Advanced features" of the settings panel.

I like the idea of importing self-trained BirdNET / Chirpity models too. It's a natural thought that if you correct incorrect detections (repeatedly) you could use that audio to build a set of negative examples to fine-tune the models. I may need to pick @Kahst's brains on that one, as in my ignorance, I am not sure if it is possible to train a new model effectively if you do not have access to at least a few dozen examples of every class you want the tuned model to recognise. It would be perfect if you could just "add a record" to the model and have it incrementally adapt, but to the best of my knowledge, it doesn't work that way.

However, let's say you trained a model on your local birds or noises and have created a training set for them. So long as the inputs had the right shape, I don't see why it would be *so* difficult to allow their import in Chirpity. It would just need custom lists to allow you to swap out the labels for the new classes.

And custom lists is quite likely what I will work on next - after looking at a filter to exclude birds that never call at night from the detections in nocturnal recordings.

**Comment by Vollpflock** [comment]
https://github.com/Mattk70/Chirpity-Electron/issues/28 I would like to see it applied on export as well, which is why I asked for it to be labeled as an "Experimental feature".

**Comment by Mattk70** [comment]
HI - I have now released an update which allows you to adjust the playback volume in the application.

**Reply by FlorianH27** [reply]
Thank you very much! I am impressed by how quickly you implement the suggestions!
